rootfs/etc/nginx/lua/sticky.lua

```diff
@@ -0,0 +1,119 @@
+local cipher = require("cipher")
+local util = require("util")
+local string = require("resty.string")
+local sha1_crypto = require("resty.sha1")
+local md5_crypto = require("resty.md5")
+
+local sticky_sessions = ngx.shared.sticky_sessions
+
+local DEFAULT_SESSION_COOKIE_NAME = "route"
+-- Currently STICKY_TIMEOUT never expires
+local STICKY_TIMEOUT = 0
+
+local _M = {}
+
+local sha1 = sha1_crypto:new()
+if not sha1 then
+  ngx.say("failed to create the sha1 object")
+  return
+end
+
+local md5 = md5_crypto:new()
```

Comment:
don't you have to do this every time you create a new digest?

Reply:
No. md5:Final() resets everything.

Reply:
hmm it does not read like that...
can you point me to where/how you confirmed that?